src/aa_remove_data/pb_utils.py

```diff
@@ -0,0 +1,164 @@
+        """
+        data = data.replace(b"\x1b", b"\x1b\x01")  # Escape escape character
+        data = data.replace(b"\x0a", b"\x1b\x02")  # Escape newline
+        data = data.replace(b"\x0d", b"\x1b\x03")  # Escape carriage return
+        return data
+
+    def _unescape_data(self, data: bytes) -> bytes:
+        """Replace escape character alternatives with the escape characters.
+        Should exactly mirror _escape_data().
+
+        Args:
+            data (bytes): A serialised protobuf message with escape characters
+            replaced.
+
+        Returns:
+            bytes: The serialised protobuf message containing escape
+            characters.
+        """
+        data = data.replace(b"\x1b\x03", b"\x0d")  # Unescape carriage return
+        data = data.replace(b"\x1b\x02", b"\x0a")  # Unescape newline
+        data = data.replace(b"\x1b\x01", b"\x1b")  # Unescape escape character
+        return data
+
+    def _convert_to_class_name(self, sample_type: str) -> str:
+        """Convert the name of a sample type to CamelCase to correspond to
+        that sample type's class name.
+
+        Args:
+            sample_type (str): Name of sample type.
+
+        Returns:
+            str: Name of sample class, e.g VectorDouble.
+        """
+        # Split the enum name by underscores and capitalize each part
+        parts = sample_type.split("_")
+        return "".join(part.capitalize() for part in parts)
+
+    def convert_to_datetime(self, year: int, seconds: int) -> datetime:
+        """Get the date and time from a year and the number of seconds passed.
+        Args:
+            year (int): A year
+            seconds (int): The number of seconds into that year that have
+            passed.
+        Returns:
+            datetime: A datetime object of the correct date and time.
+        """
+        return datetime(year, 1, 1) + timedelta(seconds=seconds)
+
+    def get_datastr(self, sample: type, year: int) -> str:
+        """Get a string contaiing information about a sample.
+        Args:
+            sample (type): A sample from a pb file.
+            year (int): The year the sample was collected.
+        Returns:
+            str: A string containing the sample information.
+        """
+        date = self.convert_to_datetime(year, sample.secondsintoyear)
+        return (
+            f"{date}    {sample.secondsintoyear:8d}    {sample.nano:9d}"
+            f"    {sample.val}\n"
+        )
+
+    def get_sample_type(self) -> str:
+        """Get the name of a pb file's sample type using information in its
+        header.
+
+        Returns:
+            str: Name of sample type, e.g VECTOR_DOUBLE.
+        """
+        type_descriptor = self.header.DESCRIPTOR.fields_by_name["type"]
+        enum_descriptor = type_descriptor.enum_type
+        return enum_descriptor.values_by_number[self.header.type].name
+
+    def get_sample_class(self) -> type:
```

Comment:
Sample doesn't feel right here either. This is getting information from the python API or schema object generated from the .proto file.
Again, I'm just throwing out suggestions (and perhaps we can have a chat about this), but maybe we just call it `proto` or `python_proto` or even `schema` to be more in line with what it represents.

Reply:
This function has been changed to `get_proto_class`.